model_analyzer/plots/plot_manager.py

```diff
@@ -12,85 +12,175 @@
     """
-    def __init__(self, config):
+    def __init__(self, config, result_manager):
         """
         Parameters
         ----------
-        config : AnalyzerConfig
+        config : ConfigCommandProfile
             The model analyzer's config containing information
             about the kind of plots to generate
+        result_manager : ResultManager
+            instance that manages the result tables and
+            adding results
         """
 
         self._config = config
+        self._result_manager = result_manager
 
         # Construct plot output directory
         self._plot_export_directory = os.path.join(config.export_path, 'plots')
         os.makedirs(self._plot_export_directory, exist_ok=True)
 
+        # Dict of list of plots
+        self._simple_plots = defaultdict(list)
+        self._detailed_plots = {}
+
+    def create_summary_plots(self):
+        """
+        Constructs simple plots based on config specs
+        """
+
         # Constraints should be plotted as well
         self._constraints = ConstraintManager.get_constraints_for_all_models(
             self._config)
 
-        # Dict of list of plots
-        self._plots = defaultdict(list)
+        model_names = [
+            model.model_name() for model in self._config.analysis_models
+        ]
 
-    def init_plots(self, plots_key):
+        for plots_key in model_names:
+            self._create_summary_plot_for_model(
+                plots_key=plots_key,
+                model_name=plots_key,
+                num_results=self._config.num_configs_per_model)
+
+        if self._config.num_top_model_configs:
+            self._create_summary_plot_for_model(
+                plots_key=TOP_MODELS_REPORT_KEY,
+                model_name=None,
+                num_results=self._config.num_top_model_configs)
+
+    def _create_summary_plot_for_model(self, model_name, plots_key,
+                                       num_results):
         """
-        Constructs new plots based on config
+        helper function that creates the summary plots
+        for a given model
         """
 
-        self._plots[plots_key] = [
-            Plot(name=plot.name(),
-                 title=plot.title(),
-                 x_axis=plot.x_axis(),
-                 y_axis=plot.y_axis(),
-                 monotonic=plot.monotonic()) for plot in self._config.plots
-        ]
+        for plot_config in self._config.plots:
+            constraints = self._constraints['default']
+            if plots_key in self._constraints:
+                constraints = self._constraints[plots_key]
+            for result in self._result_manager.top_n_results(
+                    model_name=model_name, n=num_results):
+                if result.model_config().cpu_only():
+                    if plot_config.y_axis() == 'gpu_used_memory':
+                        plot_name, plot_config_dict = list(
+                            DEFAULT_CPU_MEM_PLOT.items())[0]
```

Comment:
Because the plots for the `analyze` subcommand are not configurable by the user and are specified on a global basis. Making it per model, but not configurable was difficult.